packages/grid/x-data-grid/src/components/cell/GridEditDateCell.tsx

```diff
@@ -11,6 +12,10 @@ import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
 
 type OwnerState = { classes: DataGridProcessedProps['classes'] };
 
+const StyledInputBase = styled(InputBase)({
+  fontSize: 'inherit',
+});
```

Comment:
Why not use font-size inherit, but it feels inconsistent with how we handle the other input types:

https://github.com/mui/mui-x/blob/119590d7401019e2a741fa4b6fd9bd4d570b1f8d/packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx#L27-L39

 I would assume that we miss the slot and overridesResolver. We already have the class name for global overrides. So in the end, we might be missing an abstraction for handling text fields, of different types, shared between data picker, number, text, etc, types.

I have opened https://github.com/mui/mui-x/issues/5513 so we can push the solution to the far end.